### ISSUE TITLE
Fail-fast on environment variable not set

### ISSUE THREAD

**Issue by dkirrane** [GoogleContainerTools/skaffold#2562]
### Expected behavior

Fail fast if an environment variable is not set.

### Actual behavior

**\<no value\>** because `VERSION` environment variable is not set.

```
$ skaffold run
?[34mGenerating tags...?[0m
?[34m - eu.gcr.io/oceana-team/ip-office-poc -> ?[0meu.gcr.io/my-project/my-app:<no value>
```

_sometime later ... after jib build:_

**Invalid image reference: eu.gcr.io/my-project/my-app:<no value>**
```
[ERROR] Failed to execute goal com.google.cloud.tools:jib-maven-plugin:1.4.0:build (default-cli) on project my-app: Invalid image reference eu.gcr.io/my-project/my-app:<no value>, perhaps you should check that the reference is formatted correctly according to https://docs.docker.com/engine/reference/commandline/tag/#extended-description
[ERROR] For example, slash-separated name components cannot have uppercase letters: Invalid image reference: eu.gcr.io/my-project/my-app:<no value>
```

### Information

- Skaffold version: v0.34.1
- Operating system: Windows 10 64-bit PC
- Contents of skaffold.yaml: 

```yaml
apiVersion: skaffold/v1beta13
kind: Config
build:
  artifacts:
  - image: eu.gcr.io/my-project/my-app
    context: ./app
    jibMaven:
      args:
      - -Djib.httpTimeout=300000
      - -Dmaven.test.skip=true
  tagPolicy:
    envTemplate:
      template: '{{.IMAGE_NAME}}:{{.VERSION}}'
```

### Steps to reproduce the behavior

1. `skaffold run`

**Comment by balopat** [comment]
I think this is great to fail fast in general, and this would be relatively simple to implement. 
However if we have users who want to have "empty var" in certain environments, that can be problematic on Windows for example where creating an empty variable means deleting the variable as far as I know.

**Comment by balopat** [comment]
We discussed this - let's get a design proposal in for this - there are a bunch of options how we can implement this. Our current thinking is that this should be optional behavior - some questions to explore:
- should this apply to all templated fields (https://skaffold.dev/docs/how-tos/templating/) or just this template.  
- what should be the name of the field: `failFast:true` or `allowEmpty:false` or `verifyTemplate: true` or similar 
@dkirrane  would you be up for writing up a design doc based on our [Design proposal process](https://github.com/GoogleContainerTools/skaffold/blob/master/docs/design_proposals/README.md) to discuss?

**Comment by balopat** [comment]
As @dgageot pointed out in #2793 (and as I'm rereading the original post it is obvious that) I was completely wrong about this - the current version rendered unset env vars to `<no value>` - hence there is no way people depended on this behavior. @dgageot implemented this in #2793 - closing the issue.

**Comment by gentunian** [comment]
I'm surprized about why the decision of setting "<no value>" value to an empty environment variable! That is a side effect nobody wants. How do you check against unset environment variables? Checking if a variable differs from `<no value>` doesn't look quite right / portable. I would expect that if an environment variable was not set to keep it that way.
